### PR TITLE
qt: Add shortcuts and status bar widgets to toggle and set 3D factor

### PR DESCRIPTION
**Pull request by Steveice10**
Revival of https://github.com/citra-emu/citra/pull/4787

Updated to:
* Rebase on the current citra.
* Account for more recent changes to the 3D setting, supporting toggling between all of the available 3D modes.
* Update the UI a bit to make the interactivity of the status bar element more clear. Similar to the API indicator in the Vulkan branch, the 3D option is a push button on the left side of the status bar. This should fit in better with that when it is merged with Vulkan.
    * Backported some of the Qt stylesheet changes from the Vulkan PR to allow this.

UI Preview:

Off:
Screenshot 2023-02-02 at 5 39 50 AM

Any other state:
Screenshot 2023-02-02 at 5 39 58 AM

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6277)

### CODE REVIEW

**Review by GPUCode — APPROVED** [review]

**Comment by GPUCode** [comment]
Judging from the screenshot you might need to increase the margin a bit, the nob looks cut off from the top

**Comment by Steveice10** [comment]
Added padding:
Screenshot 2023-02-02 at 7 35 28 AM

**Comment by GPUCode** [comment]
Here's how it looks on KDE for reference
image
image

**Comment by FearlessTobi** [comment]
Untagging because of merge conflicts.

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -2453,8 +2522,16 @@ void GMainWindow::UpdateUITheme() {
...
-        qApp->setStyleSheet({});
-        setStyleSheet({});
+        const QString theme_uri(QStringLiteral(":default/style.qss"));
+        QFile f(theme_uri);
+        if (f.open(QFile::ReadOnly | QFile::Text)) {
+            QTextStream ts(&f);
+            qApp->setStyleSheet(ts.readAll());
+            setStyleSheet(ts.readAll());
+        } else {
+            qApp->setStyleSheet({});
```
Comment (FearlessTobi) [inline src/citra_qt/main.cpp:2532]: we should log an error here if loading the stylesheet fails.

**Comment by Steveice10** [comment]
Rebased and added requested log message

**Comment by Steveice10** [comment]
Rebased

**Pull request merged**